### PR TITLE
Add new CreateTabletReq and AlterTabletReqV2 thrift struct

### PR DESCRIPTION
**Pull request by morningman**
This is for developing new alter table process.

### CODE REVIEW

File: gensrc/thrift/AgentService.thrift [hunk 1]
```diff
@@ -48,6 +48,8 @@ struct TCreateTabletReq {
     4: optional Types.TVersionHash version_hash
     5: optional Types.TStorageMedium storage_medium
     6: optional bool in_restore_mode
+    // this new tablet should be colocate with base tablet
+    7: optional Types.TTabletId base_tablet_id
```
Comment (yiguolei) [inline gensrc/thrift/AgentService.thrift]: add base schema hash here

File: gensrc/thrift/AgentService.thrift [hunk 2]
```diff
@@ -61,6 +63,18 @@ struct TAlterTabletReq{
...
+// This v2 request will replace the old TAlterTabletReq.
+// TAlterTabletReq should be deprecated after new alter job process merged.
+struct TAlterTabletReqV2 {
+    1: required Types.TTabletId base_tablet_id
+    2: required Types.TTabletId new_tablet_id
+    3: required Types.TSchemaHash base_schema_hash
+    4: required Types.TSchemaHash new_schema_hash
+    // version of data which this alter task should transform
+    5: optional Types.TVersion alter_version
+    6: optional Types.TVersion alter_version_hash
```
Comment (yiguolei) [inline gensrc/thrift/AgentService.thrift]: version hash?

**Review by yiguolei — CHANGES_REQUESTED** [review]
some comments

**Review by yiguolei — COMMENTED** [review]
LGTM

**Review by imay — APPROVED** [review]

**Pull request merged**